### PR TITLE
Canavandl/uievents testing

### PR DESCRIPTION
**Pull request by canavandl**
cleans up `UIEvents._trigger` special case handling (i.e. clicking on legends)
Adds tests for 
* UIEvents._trigger
* UIEvents._bokify_hammer
* UIEvents._bokify_point_event
* All applicable UIEvents._event methods

### CODE REVIEW

File: bokehjs/test/common/ui_events.coffee [hunk 1]
```diff
@@ -1,59 +1,209 @@
...
+
+    beforeEach ->
+      @spy_trigger = sinon.spy(@ui_events, "trigger")
+
+    describe "base_type=move", ->
+
+      beforeEach ->
+        @e = new Event("move")
+        @e.bokeh = {}
+        @e.isTrusted = false
```
Comment (mattpap) [inline bokehjs/test/common/ui_events.coffee]: ?

Comment (canavandl) [inline bokehjs/test/common/ui_events.coffee]: Just testing cruft. I've removed it.

Comment (mattpap) [inline bokehjs/test/common/ui_events.coffee]: OK. I was just curious, well, still I am. How is this useful?

File: bokehjs/src/coffee/models/tools/inspectors/inspect_tool.coffee [hunk 2]
```diff
@@ -8,13 +8,3 @@ export class InspectTool extends ButtonTool
   @override {
     active: true
   }
-
```
Comment (canavandl) [inline bokehjs/src/coffee/models/tools/inspectors/inspect_tool.coffee]: deadcode removal

File: bokehjs/src/coffee/models/tools/inspectors/inspect_tool.coffee [hunk 3]
```diff
@@ -8,13 +8,3 @@ export class InspectTool extends ButtonTool
...
-  bind_bokeh_events: () ->
-    super()
-    @listenTo(events, 'move', @_inspect)
-
-  _inspect: (vx, vy, e) ->
-
-  _exit_inner: () ->
-
-  _exit_outer: () ->
```
Comment (canavandl) [inline bokehjs/src/coffee/models/tools/inspectors/inspect_tool.coffee]: dead code removal

File: bokehjs/src/coffee/models/tools/gestures/poly_select_tool.coffee [hunk 4]
```diff
@@ -25,26 +25,18 @@ export class PolySelectToolView extends SelectToolView
     @_clear_data()
 
   _clear_data: () ->
-    @data = null
+    @data = {vx: [], vy: []}
```
Comment (canavandl) [inline bokehjs/src/coffee/models/tools/gestures/poly_select_tool.coffee]: Clean up for testing reasons

**Comment by canavandl** [comment]
ping @mattpap for a quick review if you're able, otherwise I'll merge today/tomorrow

**Comment by mattpap** [comment]
@canavandl, I will review later today.

**Comment by mattpap** [comment]
I tested this on `models/file/synthetic.py` example and everything seem too work.

**Pull request merged**